template.tpl

```diff
@@ -426,18 +458,22 @@ const updateGCMState = (eventData) => {
   if (eventData === 'ready' && isDidomiConsentUnknown(didomiState, 'ad_storage') && isDidomiConsentUnknown(didomiState, 'analytics_storage')) return;
 
   const statusFromDidomiState = {};
+  
+  let consentTypes;
 
-  if (!isDidomiConsentUnknown(didomiState, 'ad_storage')) {
-    statusFromDidomiState.ad_storage = getGCMPurposeStatus(isDidomiConsentGranted(didomiState, 'ad_storage'));
+  if (data.enableTCFAdvertiserConsentMode) {
+    consentTypes = nonAdvertisingConsentTypes;
+  } else {
+    consentTypes = Object.keys(GCM_PURPOSES_MAP);
   }
 
-  if (!isDidomiConsentUnknown(didomiState, 'analytics_storage')) {
```

Comment:
This was not the right. Analytics should have been isolated and associated with the google analytics vendors. Fixed with the introduction of `consentTypesVendorMap`